### ISSUE TITLE
Polygon issues on IE11

### ISSUE THREAD

**Issue by Greigrm** [Leaflet/Leaflet.draw#618]
Seeing a couple of issues with IE11 and the polygon feature.  Release 0.4.1 / Leaflet 1.0

Firstly - intersections are constantly warned about (incorrectly) - can be seen in github example.
Secondly - you can never click the first point to complete the shape.

**Comment by ddproxy** [comment]
Notes:

1. Something is not binding?
2. Related to a touch handler

I'll look into this, thanks

**Comment by WangJi** [comment]
Draw Polygon fails on Edge too.
It's hard to click on the first point to close to polygon, 
The node-icon displays in Edge is in 20*20 px size, while by default it should be 8*8.
![image](https://cloud.githubusercontent.com/assets/5742670/19689515/3e057d66-9b00-11e6-83a7-d6ff15ada3b1.png)

**Comment by WangJi** [comment]
```L.Browser.touch``` 
is a ugly fix

**Comment by ddproxy** [comment]
I agree this is an issue with touch. Ugly fix too, but this is severely browser dependent.

**Comment by Greigrm** [comment]
Yes, setting L.Browser.touch to false only under IE works for me - thanks.  Happy for you to close this.

**Comment by richardhinkamp** [comment]
Setting `L.Browser.touch = false;` gives normal sized node icon but closing the polygon is still nearly impossible. Same for ending a polyline by clicking on the last node.

I'm using leaflet 1.0.2 and draw 0.4.7.

Having this issue in IE11 and Edge, also with Safari on iPad. 

With Chrome on Android it closes the polygon, but also adds an extra node close the starting node, so that seems to do both adding a node and closing the polygon.

**Comment by ddproxy** [comment]
Are you targeting touch devices or both?

**Comment by richardhinkamp** [comment]
We have users with ipads, android tablets, all desktop browsers and all desktop browsers with touchscreens, so pretty much everything. We might get away with disabling draw capabilities for touchscreens on desktop browsers, forcing them to use the mouse.

**Comment by B3nCr** [comment]
Forcing them to use the mouse might not fix the issue.

Back in 4.3 we had users with touch screen laptops that only used mouse and
they still had the issue and also users that didn't have touch screens and
still had the issue in IE.

**Comment by richardhinkamp** [comment]
Well since `L.Browser.touch` is always true in IE, even with no touchscreen, that makes sense.

**Comment by richardhinkamp** [comment]
I've been doing a little digging. When enabling a `L.Draw.Polygon` on the first click 2 touch events and 1 mouse up event occur. So it adds 3 markers to the map at the same location. Clicking on that marker will click the one at the top, which is the 3rd and so you can never click the first marker. This is in IE11 which behaves as pointer AND touch. Will need to check what happens on ipad, maybe just 2 touch events which results in the same problem.

So there is a bug in the event handling, but maybe draw can get smarter too. When drawing a polyline or polygon, multiple consecutive markers at the same location make no sense and could be filtered?

**Comment by B3nCr** [comment]
That's basically what my PR addressed. I checked to see if the event was
handled and then didn't handle it more than once.

**Comment by germanjoey** [comment]
@Richard Hinklamp - I'm honestly surprised that the code that checks for
poly self-intersections doesn't prevent multiple consecutive markers
already.

**Comment by germanjoey** [comment]
Oh, interesting, it seems that the poly intersection check only happens
when editing a polygon/polyline, not when drawing a new one. That shouldn't
be too hard to fix, lemme see if I can do that right now.

**Comment by germanjoey** [comment]
What does this comment mean? I could perhaps fix this but I need to know
what "degenerate cases" means.

// @method newLatLngIntersects(): boolean
// Check for intersection if new latlng was added to this polyline.
// *NOTE: does not support detecting intersection for degenerate cases.*

    // @method newPointIntersects(): boolean
// Check for intersection if new point was added to this polyline.
// newPoint must be a layer point.
// *NOTE: does not support detecting intersection for degenerate cases.*

**Comment by ddproxy** [comment]
So, this is legacy... old... detection code. Apparently the algorithm used doesn't detect some complex cases - which I can only imagine as spirals?

**Comment by richardhinkamp** [comment]
I got a simple fix for not adding double markers, which fixes some issues. iPad works, IE has still some issues because mouse+touch events. I noticed some more weird event stuff, 1 mousedown followed by a double mouseup, double touch events. I guess broken events should be fixed first. I'll see if I can find anything.

**Comment by richardhinkamp** [comment]
Double events seem to be a leaflet problem with `PointerEvent`, being worked on: https://github.com/Leaflet/Leaflet/issues/5180 This occurs in IE11 and Chrome 55+. When that's fixed, this issue can (hopefully) be properly fixed.

**Comment by z0d14c** [comment]
I have a fix for the "difficulty closing the polyline/polygon" issue in my PR: https://github.com/Leaflet/Leaflet.draw/pull/654

Unfortunately, the duplicated pointerevent/clicks/touches issue persists. I have a fork of the 0.4.3 version of Leaflet-draw where this does not occur that we've been using in production, so it must be some changed in Leaflet/Leaflet-draw that has occurred since then (possibly in conjunction with chrome updates) which causes these issues.

**Comment by cosme-benito** [comment]
Any updates regarding the oversized square node icons? 

I am aware that setting L.Browser.touch = false; might fix it but we have users with touch devices so this is not really desired...

**Comment by mlazowik** [comment]
Some of these got resolved in #661, right?
